app/controllers/users_controller.rb

```diff
@@ -1,5 +1,11 @@
 class UsersController < ApplicationController
-  def index; end
+  def index
+    @users = User.all
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:id])
+    @user_posts = Post.where(author_id: @user.id)
+    @recent_posts = @user.get_recent_posts
```

Comment:
- [ ] You have this `recent_posts` here. But in the user show view page you are still calling `user.get_recent_posts` again. Let's try to eliminate unnecessary queries to save time and clean up our code. Kindly remove this line, or use the `@recent_posts` variable in the view instead.